projects/packages/publicize/src/class-publicize-base.php

```diff
@@ -1475,6 +1476,38 @@ public function publicize_checkbox_default( $checked, $post_id ) {
 		return $checked;
 	}
 
+	/**
+	 * Checks whether sig is enabled for a post.
+	 *
+	 * @param int $post_id ID of the current post.
+	 * @return bool True if sig is enabled for the post.
+	 */
+	private function is_sig_enabled_for_post( $post_id ) {
+		$social_options = get_post_meta( $post_id, '_wpas_options', true );
+		return $social_options['image_generator_settings']['enabled'];
+	}
+
+	/**
+	 * Adds the sig image to the media array.
+	 *
+	 * @param array $media Array of media.
+	 * @param int   $post_id ID of the current post.
+	 * @return array Filtered array of media.
+	 */
+	public function get_sig_image_for_post( $media, $post_id ) {
+		if ( $this->is_sig_enabled_for_post( $post_id ) ) {
+			$sig       = new Social_Image_Generator();
+			$sig_image = array(
+				'type'       => 'image',
+				'src'        => $sig->get_image_url( $post_id ),
```

Comment:
I've [slightly changed this method in the library](https://github.com/Automattic/jetpack/pull/29118) so we'll need to update this, but I think it's probably best to hold off on updating this until that PR is merged 👍🏼

Reply:
Noted!